modules/@ergonode/products/src/components/Tabs/ProductTemplateTab.vue

```diff
@@ -204,6 +174,30 @@ export default {
         this.language = this.languageOptions
             .find(languageCode => languageCode.code === this.defaultLanguageCode);
     },
+    mounted() {
```

Comment:
I would transfer to a new component ex. Observer.vue
ex. 
`<template>
    <div class="observer">
        <slot />
    </div>
</template>

<script>
export default {
    name: 'Observer',
    props: {
        options: {
            type: Object,
            default: () => ({}),
        },
        observeOnce: {
            type: Boolean,
            default: false,
        },
    },
    data() {
        return {
            observer: null,
        };
    },
    mounted() {
        this.observer = new IntersectionObserver(([entry]) => {
            if (!entry) {
                return;
            }

            this.$emit('intersect', entry.isIntersecting);

            if (entry.isIntersecting && this.observeOnce) {
                this.observer.disconnect();
                this.observer = null;
            }
        }, this.options);

        this.observer.observe(this.$el);
    },
    beforeDestroy() {
        if (this.observer) {
            this.observer.disconnect();
        }
    },
};
</script>
`
will be more reusable